asv_bench/benchmarks/attrs_caching.py

```diff
@@ -1,5 +1,5 @@
 from .pandas_vb_common import *
-from pandas.util.decorators import cache_readonly
```

Comment:
You can do the public import here. But, you will also have to try the older import paths.

The hashing imports also can use an update I think